server/plugins/api/GenericNotifier.ts

```diff
@@ -0,0 +1,8 @@
+import { GenericConfig } from "./GenericConfig"
+import { APIReturn } from "./APIReturn"
+
+
+export interface GenericNotifier {
+    sendMessage<T extends GenericConfig>(config: T) : APIReturn
```

Comment:
Maybe it would be cleaner to get rid of `GenericConfig` and just have this method be like:
```typescript
sendMessage<T>(message: String, config: T): APIReturn;
```
This way each plugin can have whatever they want for a config and we know there's always going to be a message, and there's less interfaces involved.